In [ ]:
# 원래 코드
# -*- coding: utf-8 -*-
# 코드 내부에 한글을 사용가능 하게 해주는 부분입니다.

# 딥러닝을 구동하는 데 필요한 케라스 함수를 불러옵니다.
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense

# 필요한 라이브러리를 불러옵니다.
import numpy as np
import tensorflow as tf

# 실행할 때마다 같은 결과를 출력하기 위해 설정하는 부분입니다.
np.random.seed(3)
tf.random.set_seed(3)

# 준비된 수술 환자 데이터를 불러들입니다.
Data_set = np.loadtxt("/content/drive/MyDrive/colab/deeplearning/dataset/ThoraricSurgery.csv", delimiter=",")

# 환자의 기록과 수술 결과를 X와 Y로 구분하여 저장합니다.
X = Data_set[:,0:17]
Y = Data_set[:,17]

test_X = X[:5, :]
test_Y = Y[:5]

# 딥러닝 구조를 결정합니다(모델을 설정하고 실행하는 부분입니다).
model = Sequential()
model.add(Dense(30, input_dim=17, activation='relu'))
model.add(Dense(1, activation='sigmoid'))

# 딥러닝을 실행합니다.
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
model.fit(X, Y, epochs=100, batch_size=10)

#예측 함수
predictions = model.predict(test_X)

# 실제 값, 예측 값 출력
print("Actual Class: ", test_Y)
print("Predict Class:" )
for i in range(5):
  print("%0.4f" % predictions[i])

Epoch 1/100
47/47 [==============================] - 1s 2ms/step - loss: 1.2715 - accuracy: 0.6851
Epoch 2/100
47/47 [==============================] - 0s 2ms/step - loss: 0.5881 - accuracy: 0.8213
Epoch 3/100
47/47 [==============================] - 0s 2ms/step - loss: 0.5149 - accuracy: 0.8213
Epoch 4/100
47/47 [==============================] - 0s 2ms/step - loss: 0.5488 - accuracy: 0.8404
Epoch 5/100
47/47 [==============================] - 0s 2ms/step - loss: 0.4900 - accuracy: 0.8511
Epoch 6/100
47/47 [==============================] - 0s 2ms/step - loss: 0.4685 - accuracy: 0.8511
Epoch 7/100
47/47 [==============================] - 0s 2ms/step - loss: 0.4864 - accuracy: 0.8553
Epoch 8/100
47/47 [==============================] - 0s 2ms/step - loss: 0.4871 - accuracy: 0.8489
Epoch 9/100
47/47 [==============================] - 0s 2ms/step - loss: 0.4424 - accuracy: 0.8532
Epoch 10/100
47/47 [==============================] - 0s 2ms/step - loss: 0.4590 - accuracy: 0.8489
Epoch 11/

<ipython-input-22-ab626c733136>:42: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
  print("%0.4f" % predictions[i])


In [ ]:
# epoch값 1000
# -*- coding: utf-8 -*-
# 코드 내부에 한글을 사용가능 하게 해주는 부분입니다.

# 딥러닝을 구동하는 데 필요한 케라스 함수를 불러옵니다.
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense

# 필요한 라이브러리를 불러옵니다.
import numpy as np
import tensorflow as tf

# 실행할 때마다 같은 결과를 출력하기 위해 설정하는 부분입니다.
np.random.seed(3)
tf.random.set_seed(3)

# 준비된 수술 환자 데이터를 불러들입니다.
Data_set = np.loadtxt("/content/drive/MyDrive/colab/deeplearning/dataset/ThoraricSurgery.csv", delimiter=",")

# 환자의 기록과 수술 결과를 X와 Y로 구분하여 저장합니다.
X = Data_set[:,0:17]
Y = Data_set[:,17]

test_X = X[:5, :]
test_Y = Y[:5]

# 딥러닝 구조를 결정합니다(모델을 설정하고 실행하는 부분입니다).
model = Sequential()
model.add(Dense(30, input_dim=17, activation='relu'))
model.add(Dense(1, activation='sigmoid'))

# 딥러닝을 실행합니다.
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
model.fit(X, Y, epochs=1000, batch_size=10)

#예측 함수
predictions = model.predict(test_X)

# 실제 값, 예측 값 출력
print("Actual Class: ", test_Y)
print("Predict Class:" )
for i in range(5):
  print("%0.4f" % predictions[i])

Epoch 1/1000
47/47 [==============================] - 1s 2ms/step - loss: 4.0254 - accuracy: 0.8447
Epoch 2/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.8790 - accuracy: 0.7787
Epoch 3/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.6522 - accuracy: 0.8021
Epoch 4/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.5858 - accuracy: 0.8213
Epoch 5/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.5331 - accuracy: 0.8362
Epoch 6/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4921 - accuracy: 0.8362
Epoch 7/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4887 - accuracy: 0.8447
Epoch 8/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4745 - accuracy: 0.8404
Epoch 9/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4460 - accuracy: 0.8511
Epoch 10/1000
47/47 [==============================] - 0s 2ms/step - loss: 0.4669 - accuracy: 0.8426

<ipython-input-27-3d9519f432ba>:42: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
  print("%0.4f" % predictions[i])


In [ ]:
# batch_size 94로 증가
# -*- coding: utf-8 -*-
# 코드 내부에 한글을 사용가능 하게 해주는 부분입니다.

# 딥러닝을 구동하는 데 필요한 케라스 함수를 불러옵니다.
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense

# 필요한 라이브러리를 불러옵니다.
import numpy as np
import tensorflow as tf

# 실행할 때마다 같은 결과를 출력하기 위해 설정하는 부분입니다.
np.random.seed(3)
tf.random.set_seed(3)

# 준비된 수술 환자 데이터를 불러들입니다.
Data_set = np.loadtxt("/content/drive/MyDrive/colab/deeplearning/dataset/ThoraricSurgery.csv", delimiter=",")

# 환자의 기록과 수술 결과를 X와 Y로 구분하여 저장합니다.
X = Data_set[:,0:17]
Y = Data_set[:,17]

test_X = X[:5, :]
test_Y = Y[:5]

# 딥러닝 구조를 결정합니다(모델을 설정하고 실행하는 부분입니다).
model = Sequential()
model.add(Dense(30, input_dim=17, activation='relu'))
model.add(Dense(1, activation='sigmoid'))

# 딥러닝을 실행합니다.
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
model.fit(X, Y, epochs=100, batch_size=94)

#예측 함수
predictions = model.predict(test_X)

# 실제 값, 예측 값 출력
print("Actual Class: ", test_Y)
print("Predict Class:" )
for i in range(5):
  print("%0.4f" % predictions[i])

Epoch 1/100
5/5 [==============================] - 1s 5ms/step - loss: 2.1360 - accuracy: 0.6489
Epoch 2/100
5/5 [==============================] - 0s 3ms/step - loss: 1.6465 - accuracy: 0.6489
Epoch 3/100
5/5 [==============================] - 0s 4ms/step - loss: 1.2326 - accuracy: 0.7021
Epoch 4/100
5/5 [==============================] - 0s 3ms/step - loss: 1.0079 - accuracy: 0.7596
Epoch 5/100
5/5 [==============================] - 0s 4ms/step - loss: 0.8627 - accuracy: 0.7851
Epoch 6/100
5/5 [==============================] - 0s 4ms/step - loss: 0.8255 - accuracy: 0.7809
Epoch 7/100
5/5 [==============================] - 0s 4ms/step - loss: 0.7828 - accuracy: 0.7957
Epoch 8/100
5/5 [==============================] - 0s 4ms/step - loss: 0.7516 - accuracy: 0.7745
Epoch 9/100
5/5 [==============================] - 0s 4ms/step - loss: 0.7417 - accuracy: 0.8043
Epoch 10/100
5/5 [==============================] - 0s 4ms/step - loss: 0.7131 - accuracy: 0.7766
Epoch 11/100
5/5 [===========

<ipython-input-26-11c6c81a1b7a>:42: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
  print("%0.4f" % predictions[i])


In [ ]:
# batch_size 94로 증가 & epoch값 1000
# -*- coding: utf-8 -*-
# 코드 내부에 한글을 사용가능 하게 해주는 부분입니다.

# 딥러닝을 구동하는 데 필요한 케라스 함수를 불러옵니다.
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense

# 필요한 라이브러리를 불러옵니다.
import numpy as np
import tensorflow as tf

# 실행할 때마다 같은 결과를 출력하기 위해 설정하는 부분입니다.
np.random.seed(3)
tf.random.set_seed(3)

# 준비된 수술 환자 데이터를 불러들입니다.
Data_set = np.loadtxt("/content/drive/MyDrive/colab/deeplearning/dataset/ThoraricSurgery.csv", delimiter=",")

# 환자의 기록과 수술 결과를 X와 Y로 구분하여 저장합니다.
X = Data_set[:,0:17]
Y = Data_set[:,17]

test_X = X[:5, :]
test_Y = Y[:5]

# 딥러닝 구조를 결정합니다(모델을 설정하고 실행하는 부분입니다).
model = Sequential()
model.add(Dense(30, input_dim=17, activation='relu'))
model.add(Dense(1, activation='sigmoid'))

# 딥러닝을 실행합니다.
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
model.fit(X, Y, epochs=1000, batch_size=94)

#예측 함수
predictions = model.predict(test_X)

# 실제 값, 예측 값 출력
print("Actual Class: ", test_Y)
print("Predict Class:" )
for i in range(5):
  print("%0.4f" % predictions[i])

Epoch 1/1000
5/5 [==============================] - 1s 6ms/step - loss: 55.9652 - accuracy: 0.1489
Epoch 2/1000
5/5 [==============================] - 0s 5ms/step - loss: 50.2963 - accuracy: 0.1489
Epoch 3/1000
5/5 [==============================] - 0s 4ms/step - loss: 44.6696 - accuracy: 0.1489
Epoch 4/1000
5/5 [==============================] - 0s 5ms/step - loss: 39.1233 - accuracy: 0.1489
Epoch 5/1000
5/5 [==============================] - 0s 4ms/step - loss: 33.6292 - accuracy: 0.1489
Epoch 6/1000
5/5 [==============================] - 0s 4ms/step - loss: 28.1952 - accuracy: 0.1489
Epoch 7/1000
5/5 [==============================] - 0s 4ms/step - loss: 22.8419 - accuracy: 0.1489
Epoch 8/1000
5/5 [==============================] - 0s 3ms/step - loss: 17.4003 - accuracy: 0.1489
Epoch 9/1000
5/5 [==============================] - 0s 4ms/step - loss: 12.1411 - accuracy: 0.1489
Epoch 10/1000
5/5 [==============================] - 0s 5ms/step - loss: 6.8442 - accuracy: 0.1915
Epoch 11/1

<ipython-input-29-918db9d5b806>:42: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
  print("%0.4f" % predictions[i])


1. epoch:100, batch:10 - 0.401, 0.846 00000
2. epoch:1000, batch:10 - 0.302, 0.885 00110
3. epoch:100, batch:94 - 0.407, 0.851 00000
4. epoch:1000, batch:94 - 0.345, 0.866 00110

- epochs가 늘어날 수록 정확도가 늘어나고 에러값이 줄어들었다. 그리고 예측값이 맞아 떨어진다.
- batch_size값이 증가할수록 정확도가 늘어났다.